storage/innobase/btr/btr0sea.cc

```diff
@@ -1120,7 +1120,8 @@ void btr_search_drop_page_hash_index(buf_block_t *block) {
   ut_ad(!index->disable_ahi);
   ut_ad(btr_search_enabled);
 
-  ut_ad(block->page.id.space() == index->space);
+  ut_ad(index->space == FIL_NULL ||
```

Comment:
The root cause of the issue is:
  When space is discarded (ALTER TABLE ... DISCARD TABLESPACE)
  we set index->space = FIL_NULL in row_discard_tablespace()
  Then when we DROP TABLE ... we get here from
  btr_search_drop_page_hash_when_freed() => buf_page_get_gen() =>
  buf_page_free_stale() => buf_LRU_free_page()
  We can detect this case by examining
  block->page.get_space()->was_not_deleted or block->page.was_stale()
  
My proposal is to fix it in the following way:
```
ut_ad(block->page.was_stale() || block->page.id.space() == index->space);
```

Reply:
Done